pkg/schedulercache/resource_info.go

```diff
@@ -16,5 +16,71 @@ limitations under the License.
 
 package schedulercache
 
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/api/core/v1"
+)
+
 type Resource struct {
+	MilliCPU float64
+	Memory   float64
+}
+
+func EmptyResource() *Resource {
+	return &Resource{
+		MilliCPU: 0,
+		Memory:   0,
+	}
+}
+
+func (r *Resource) Clone() *Resource {
+	clone := &Resource{
+		MilliCPU: r.MilliCPU,
+		Memory:   r.Memory,
+	}
+	return clone
+}
+
+var minMilliCPU float64 = 10
+var minMemory float64 = 10 * 1024 * 1024
+
+func NewResource(rl v1.ResourceList) *Resource {
+	cpu := rl[v1.ResourceCPU]
+	mem := rl[v1.ResourceMemory]
+
+	return &Resource{
+		MilliCPU: float64(cpu.MilliValue()),
+		Memory:   float64(mem.Value()),
+	}
+}
+
+func (r *Resource) IsEmpty() bool {
+	return r.MilliCPU < minMilliCPU && r.Memory < minMemory
+}
+
+func (r *Resource) Add(rr *Resource) *Resource {
+	r.MilliCPU += rr.MilliCPU
+	r.Memory += rr.Memory
+	return r
+}
+
+func (r *Resource) Sub(rr *Resource) *Resource {
+	r.MilliCPU -= rr.MilliCPU
+	r.Memory -= rr.Memory
+	return r
+}
+
+func (r *Resource) Less(rr *Resource) bool {
+	return r.MilliCPU < rr.MilliCPU && r.Memory < rr.Memory
```

Comment:
this may return `false` for `A.Less(B)` and `B.Less(A)`, such as `A(MilliCPU:2, Memory:1)`, `B(MilliCPU:2, Memory:1)`

Reply:
I think that's what we want. When allocating resource to Pod, we need to know whether we can launch Pod with resources.

Reply:
get the point :)